File: src/main/kotlin/Character.kt
package org.example

import kotlin.math.roundToInt
import kotlin.time.times

class Character(val name:String,val MaxHitPoints:Int,val BaseDamega:Int,val BaseArmor:Int,val inventory:List<Item>) {
    fun Attack():Int{
        if (inventory.isEmpty()){
            return BaseDamega
        }else {
            return (BaseDamega + (DamageAllItem() / 10))
        }
    }

    fun Defense():List<Int>{
        return ArmorAllItem()
    }
//Have all damage from inventory if it is a weapon
    fun DamageAllItem():Int{

        var totalDamageItem = 0

        for (weapon in inventory.filter { it is Weapon } ){
            totalDamageItem +=weapon.Apply(this)
        }

        return totalDamageItem
    }

    fun ArmorAllItem():Int{

        var totalArmorItem = 0

        for (weapon in inventory.filter { it is Protection } ){
            totalArmorItem +=weapon.Apply(this)
        }

        return totalArmorItem
    }
}

File: ConsoleApp1/Character.cs
using ConsoleApp1;

public class Character(string name, int MaxHitPoints, int BaseDamage,int BaseArmor, List<IItem> _inventory)
{
    
    public int Attack()
    {
        if (_inventory.Count == 0)
        {
            return BaseDamage;
        }
        else
        {
            return BaseArmor + (DamegeAllItem(_inventory)/10);
        }
    }
    //
    int DamegeAllItem(List<IItem> _inventory)
    {
        int totalDamege = 0;
        foreach (var weapon in _inventory)
        {
            if (weapon is Weapon)
            {
                totalDamege += weapon.Apply(this);
            }
        }

        return totalDamege;
    }

    void Defend()
    {
        
    }

    void Heal(int amount)
    {
        
    }

    void ReceiveDamage(int damage)
    {
        
    }
    
}